airmozilla/starred/tests/test_views.py

```diff
@@ -227,3 +227,53 @@ def test_anonymous_starred_events(self):
         ok_(event1.title in response.content)
         ok_(event2.title in response.content)
         ok_(event3.title not in response.content)
+
+    def test_event_pagination(self):
+        url = self.home_url
+        user = self._login(username='lisa')
+
+        events = []
+
+        for i in range(25):
+            event = self.create_event("Event Title %d" % (i + 1))
+            StarredEvent.objects.create(user=user, event=event)
+            events.append(event.id)
+
+        response = self.client.get(url)
+        eq_(response.status_code, 200)
+
+        for i in range(len(events)):
+            match = 'data-id="%d"' % events[i]
+            if i < 10:
+                self.assertTrue(match in response.content, i)
+            else:
+                self.assertFalse(match in response.content, i)
+
+        ok_('href="/starred/page/2/"' in response.content)
+        ok_('href="/starred/page/0/"' not in response.content)
+
+        response = self.client.get(url + 'page/2/')
+        eq_(response.status_code, 200)
+
+        for i in range(len(events)):
+            match = 'data-id="%d"' % events[i]
+            if i >= 10 and i < 20:
+                self.assertTrue(match in response.content, i)
+            else:
+                self.assertFalse(match in response.content, i)
+
+        ok_('href="/starred/page/1/"' in response.content)
+        ok_('href="/starred/page/3/"' in response.content)
+
+        response = self.client.get(url + 'page/3/')
+        eq_(response.status_code, 200)
+
+        for i in range(len(events)):
+            match = 'data-id="%d"' % events[i]
+            if i >= 20:
+                self.assertTrue(match in response.content, i)
+            else:
+                self.assertFalse(match in response.content, i)
+
+        ok_('href="/starred/page/2/"' in response.content)
+        ok_('href="/starred/page/4/"' not in response.content)
```

Comment:
The second section checks to make sure there is a next page and a previous page. The third section is different in that it checks to make sure there is only a previous page and not a next page.